components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/RoleManagementService.java

```diff
@@ -451,6 +451,16 @@ Map<String, String> getMainRoleToSharedRoleMappingsBySubOrg(List<String> roleIds
     List<String> getAssociatedApplicationByRoleId(String roleId, String tenantDomain)
             throws IdentityRoleManagementException;
 
+    /**
+     * Retrieve the main role UUIDs corresponding to a list of shared role UUIDs.
+     *
+     * @param sharedRoleIds List of shared (sub-organization) role UUIDs for which main role UUIDs are to be found.
+     * @return List of main role UUIDs corresponding to the given shared role UUIDs.
+     * @throws IdentityRoleManagementException If an error occurs while retrieving the role mappings.
+     */
+    List<String> getMainRoleUUIDsForSharedRoles(List<String> sharedRoleIds)
```

Comment:
Let's make this default, since we are modifying an existing public interface.

Reply:
Addressed and resolved in commit: [42705b9](https://github.com/wso2/carbon-identity-framework/pull/6162/commits/42705b984942a14ec734891444d31c9d33397a6f)